doc/src/manual/methods.md

```diff
@@ -211,26 +217,30 @@ of methods:
 
 ```julia-repl
 julia> methods(+)
-# 180 methods for generic function "+":
-[1] +(x::Bool, z::Complex{Bool}) in Base at complex.jl:227
-[2] +(x::Bool, y::Bool) in Base at bool.jl:89
-[3] +(x::Bool) in Base at bool.jl:86
-[4] +(x::Bool, y::T) where T<:AbstractFloat in Base at bool.jl:96
-[5] +(x::Bool, z::Complex) in Base at complex.jl:234
-[6] +(a::Float16, b::Float16) in Base at float.jl:373
-[7] +(x::Float32, y::Float32) in Base at float.jl:375
-[8] +(x::Float64, y::Float64) in Base at float.jl:376
-[9] +(z::Complex{Bool}, x::Bool) in Base at complex.jl:228
-[10] +(z::Complex{Bool}, x::Real) in Base at complex.jl:242
-[11] +(x::Char, y::Integer) in Base at char.jl:40
-[12] +(c::BigInt, x::BigFloat) in Base.MPFR at mpfr.jl:307
-[13] +(a::BigInt, b::BigInt, c::BigInt, d::BigInt, e::BigInt) in Base.GMP at gmp.jl:392
-[14] +(a::BigInt, b::BigInt, c::BigInt, d::BigInt) in Base.GMP at gmp.jl:391
-[15] +(a::BigInt, b::BigInt, c::BigInt) in Base.GMP at gmp.jl:390
-[16] +(x::BigInt, y::BigInt) in Base.GMP at gmp.jl:361
-[17] +(x::BigInt, c::Union{UInt16, UInt32, UInt64, UInt8}) in Base.GMP at gmp.jl:398
+# 207 methods for generic function "+" from Base:
+   [1] +(x::T, y::T) where {T<:Union{Int128, Int16, Int32, Int64, Int8, UInt128, UInt16, UInt32, UInt64, UInt8}}
+     @ int.jl:87
+   [2] +(c::Union{UInt16, UInt32, UInt64, UInt8}, x::BigInt)
+     @ Base.GMP gmp.jl:529
+   [3] +(c::Union{Int16, Int32, Int64, Int8}, x::BigInt)
+     @ Base.GMP gmp.jl:535
+   [4] +(c::Union{UInt16, UInt32, UInt64, UInt8}, x::BigFloat)
+     @ Base.MPFR mpfr.jl:376
+   [5] +(c::Union{Int16, Int32, Int64, Int8}, x::BigFloat)
+     @ Base.MPFR mpfr.jl:384
+   [6] +(c::Union{Float16, Float32, Float64}, x::BigFloat)
+     @ Base.MPFR mpfr.jl:392
+   [7] +(x::Union{Dates.CompoundPeriod, Dates.Period})
+     @ Dates /Users/me/.julia/dev/julia/usr/share/julia/stdlib/v1.7/Dates/src/periods.jl:372
+   [8] +(x::Rational, y::Integer)
+     @ rational.jl:310
+   [9] +(x::T, y::Integer) where {T<:AbstractChar}
+     @ char.jl:235
+  [10] +(Da::LinearAlgebra.Diagonal, Db::LinearAlgebra.Diagonal)
+     @ LinearAlgebra /Users/me/.julia/dev/julia/usr/share/julia/stdlib/v1.7/LinearAlgebra/src/diagonal.jl:172
 ...
-[180] +(a, b, c, xs...) in Base at operators.jl:424
+ [207] +(a::Any, b::Any, c::Any, xs::Any...)
```

Comment:
Was this change intentional? I think always printing the `::Any` is a bit verbose and I am not sure it adds much information.

Reply:
I agree it's a bit verbose in cases like that. What it was meant to do was to help cases like this, where otherwise untyped arguments go missing:

<img width="977" alt="Screenshot 2021-05-26 at 09 43 01" src="https://user-images.githubusercontent.com/32575566/119670359-d35b7680-be06-11eb-85ec-685cba13c4e0.png">

Reply:
I think it would be fine to omit the `::Any` here, but I don't have any strong opinions on this.

Reply:
Here are some variants which print the module & path without a new line. (Because the new one above takes up a lot of lines compared to the old. And the types printed here are usually shorter than those in the stack traces.) I think this might be better. 

On the left I omit the `::Any`, what I don't like is that suddenly the variable name has to stand on its own, instead of being clearly secondary information to the signature. You can't answer "how many arguments?" by scanning "how many bold things?" anymore. 

On the right, I've also added bold to required keywords, shown with `cumsum`. Seems like useful information to mark them somehow, hope bold isn't confusing --- all the types are also requirements, after all.

<img width="1063" alt="Screenshot 2021-05-26 at 12 01 35" src="https://user-images.githubusercontent.com/32575566/119697648-6b188f00-be1e-11eb-91e3-c7ae8612f2e4.png">